contracts/lazyMint/LazyMintERC1155.sol

```diff
@@ -0,0 +1,606 @@
+    function verifyClaimIsValid(
+        uint256 _tokenId, 
+        uint256 _quantity,
+        bytes32[] calldata _proofs,
+        uint256 _conditionIndex,
+        MintCondition memory _mintCondition
+    )
+        internal
+        view
+    {
+        require(
+            _quantity > 0 
+                && _quantity <= _mintCondition.quantityLimitPerTransaction, 
+            "LazyMintERC1155: invalid quantity claimed."
+        );        
+        require(
+            _mintCondition.currentMintSupply + _quantity <= _mintCondition.maxMintSupply,
+            "LazyMintERC1155: exceed max mint supply."
+        );
+
+        uint256 validTimestampForClaim = mintConditions[_tokenId].nextValidTimestampForClaim[_msgSender()][_conditionIndex];
+        require(
+            validTimestampForClaim == 0 
+                || block.timestamp >= validTimestampForClaim, 
+                "LazyMintERC1155: cannot claim yet."
+        );
+
+        if (_mintCondition.merkleRoot != bytes32(0)) {
+            bytes32 leaf = keccak256(abi.encodePacked(_msgSender()));
+            require(
+                MerkleProof.verify(_proofs, _mintCondition.merkleRoot, leaf), 
+                "LazyMintERC1155: not in whitelist."
+            );
+        }
+    }
+
+    /// @dev Collects and distributes the primary sale value of tokens being claimed.
+    function collectClaimPrice(
+        MintCondition memory _mintCondition,
+        uint256 _quantityToClaim,
+        uint256 _tokenId
+    )
+        internal
+    {
+        if (_mintCondition.pricePerToken <= 0) {
+            return;    
+        }
+
+        uint256 totalPrice = _quantityToClaim * _mintCondition.pricePerToken;
+        uint256 fees = (totalPrice * feeBps) / MAX_BPS;
+
+        if(_mintCondition.currency == NATIVE_TOKEN) {
+            require(
+                msg.value == totalPrice,
+                "LazyMintERC1155: must send total price."
+            );
+        } else {
+            validateERC20BalAndAllowance(
+                _msgSender(), 
+                _mintCondition.currency,
+                totalPrice
+            );
+        }
+
+        transferCurrency(
+            _mintCondition.currency,
+            _msgSender(),
+            controlCenter.getRoyaltyTreasury(address(this)),
+            fees
+        );
+        
+        address recipient = saleRecipient[_tokenId];
+        transferCurrency(
+            _mintCondition.currency,
+            _msgSender(),
+            recipient == address(0) ? defaultSaleRecipient : recipient,
+            totalPrice - fees
+        );
+    }
+
+    /// @dev Transfers the tokens being claimed.
+    function transferClaimedTokens(
+        MintCondition memory _mintCondition,
+        uint256 _mintConditionIndex,
+        uint256 _tokenId,
+        uint256 _quantityBeingClaimed
+    )
+        internal
+    {
+        _mint(_msgSender(), _tokenId, _quantityBeingClaimed, "");
```

Comment:
violates Checks-Effects-Interactions pattern

Reply:
See 2a9e0372b06699517d111490422c09e207bbebd6

All require checks made before state changes, and all state changes made before external contract calls (potential call in `_mint`)